esphome/components/deep_sleep/deep_sleep_esp32.cpp

```diff
@@ -13,6 +13,7 @@ optional<uint32_t> DeepSleepComponent::get_run_duration_() const {
     switch (wakeup_cause) {
       case ESP_SLEEP_WAKEUP_EXT0:
       case ESP_SLEEP_WAKEUP_EXT1:
+        return this->wakeup_cause_to_run_duration_->ext1_cause;
```

Comment:
This is breaking change IMO

Reply:
I guess the H2 variant has limited usefulness for esphome as a whole at the moment since ZIgbee support is only through external components. Once/if this gains traction it is very valuable due to it's much lower power consumption compared to C6.

I'm a hardware person so I have a hard time judging the quality of the current deep sleep component. My changes certainly are hacky since the H2 differs so much from the S and C series parts. In the end it is up to people more involved with esphome to decide if the H2 can be added or if a more sensible approach would be to consider a more in-depth rework of deep_sleep first (I'm not qualified to take that on) and for me to create an external component, deep_sleep_h2, to have a working solution for that part.

Reply:
You can always import deep sleep component from this pull request as external component. Just I would do the following:
```suggestion
#if defined(USE_ESP32_VARIANT_ESP32H2)
        return this->wakeup_cause_to_run_duration_->ext1_cause;
#endif
```
and also add whole EXT1 addition under same if defined preprocessor directive and most importatly mention in docs that EXT1 is used only on ESP32-H2.

Reply:
I think I get your point now! If EXT1 wakeup duration is only available for H2 this does not break anything else. As I read the Espressif docs at least the C3 and C6 also support EXT1 wakeup so the wakeup cause would be useful which was my original thought. The day job calls but I'll get to it this weekend. TODO for me:

- Make EXT1 wakeup duration only available to H2 by preprocessor.
- Update docs that EXT1 wakeup only on H2.
- Make a new PR for the docs (github can't change source branch, only destination)
- Maybe have another look at the __init__.py

I am already using deep sleep on H2 as external component from my fork, so not in a hurry from my side.

Reply:
Problem is that EXT0 wakeup is currently(in dev) being used on supported variants with GPIO run time. Your change would result into using EXT1 run time. 

You need to do the following to make it non breaking.
```cpp
      case ESP_SLEEP_WAKEUP_EXT0:
#if defined(USE_ESP32_VARIANT_ESP32H2)
        return this->wakeup_cause_to_run_duration_->gpio_cause;
#endif
      case ESP_SLEEP_WAKEUP_EXT1:
#if defined(USE_ESP32_VARIANT_ESP32H2)
        return this->wakeup_cause_to_run_duration_->ext1_cause;
#endif
      case ESP_SLEEP_WAKEUP_GPIO:
        return this->wakeup_cause_to_run_duration_->gpio_cause;
```

Reply:
I think I might just not add the ext1_cause and use the gpio_cause instead.

I mean it is ugly from the hardware perspective since EXT0 and EXT1 are quite different (EXT1 is much more constrained) and hiding this difference from the user could cause problems later but I will experiment with it and maybe then this PR (or a new one since scope will be different) might be a trivial patch just fixing .